mitosheet/src/components/taskpanes/SnowflakeImport/SnowflakeImportTaskpane.tsx

```diff
@@ -0,0 +1,478 @@
+    mitoAPI: MitoAPI;
+    userProfile: UserProfile;
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+    analysisData: AnalysisData;
+    sheetDataArray: SheetData[];
+    selectedSheetIndex: number;
+}
+
+export type SnowflakeCredentialsValidityCheckResult = {type: 'success'} | {type: 'error', 'error_message': string}
+
+export type SnowflakeCredentials = {type: 'username/password', username: string, password: string, account: string};
+
+export type SnowflakeConnection = {
+    warehouse: string | undefined, 
+    database: string | undefined, 
+    schema: string | undefined,
+    table: string | undefined
+}
+
+export type SnowflakeConnectionAPIReturn = {
+    warehouse: string | null, 
+    database: string | null, 
+    schema: string | null,
+    table: string | null
+}
+
+export type SnowflakeQueryParams = {
+	columns: string[],
+	limit: number | undefined
+}
+
+export type SnowflakeConfigOptions = {
+	warehouses: string[],
+	databases: string[],
+	schemas: string[],
+	tables: string[]
+	columns: string[]
+}
+
+export interface SnowflakeImportParams {
+    credentials: SnowflakeCredentials,
+    connection: SnowflakeConnection,
+    query_params: SnowflakeQueryParams,
+}
+
+const getDefaultParams = (): SnowflakeImportParams | undefined => {
+    return {
+        credentials: {type: 'username/password', username: '', password: '', account: ''},
+        connection: {warehouse: undefined, database: undefined, schema: undefined, table: undefined},
+        query_params: {columns: [], limit: undefined},
+    }
+}
+
+export type AvailableSnowflakeOptionsAndDefaults = {
+    type: 'success',
+    config_options: SnowflakeConfigOptions,
+    default_connection_values: SnowflakeConnectionAPIReturn
+} | {
+    type: 'error',
+    error_message: string
+}
+
+
+/* 
+    This is the SnowflakeImport taskpane.
+*/
+const SnowflakeImportTaskpane = (props: SnowflakeImportTaskpaneProps): JSX.Element => {
+
+    //loading, edit, editApplied
+    const {params, setParams, edit} = useSendEditOnClick<SnowflakeImportParams, undefined>(
+            () => getDefaultParams(),
+            StepType.SnowflakeImport, 
+            props.mitoAPI,
+            props.analysisData,
+    )
+    
+    const [openCredentialsSection, setOpenCredentialsSection] = useState(true);
+    const [availableSnowflakeOptionsAndDefaults, setAvailableSnowflakeOptionsAndDefaults] = useState<AvailableSnowflakeOptionsAndDefaults | undefined>(undefined);
+    const [liveUpdateNumber, setLiveUpdateNumber] = useState(0) 
+
+    const getAvailableOptionsAndDefaults = (newParams: SnowflakeImportParams): void => {
+        setParams(newParams)
+        setLiveUpdateNumber(old => old + 1)
+    }
+
+    useEffect(() => {
+        if (liveUpdateNumber === 0) {
+            // Don't run on first render
+            return
+        }
+        void _getAvailableOptionsAndDefaults()
+    }, [liveUpdateNumber])
+
+    const _getAvailableOptionsAndDefaults = async () => {
```

Comment:
I think this `get` function should also not set params -- that's a weird thing for a function to do!

Reply:
Perhaps this function should be named `updateAvailableOptionsAndDefaults`?